src/main/java/io/vertx/core/impl/DeploymentManager.java

```diff
@@ -374,19 +376,68 @@ public void unregisterVerticleFactory(VerticleFactory factory) {
     return factoryList;
   }
 
+  private static URL mapToURL(String path) {
+    try {
+      File file = new File(path);
+      return file.toURI().toURL();
+    } catch (MalformedURLException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private static List<URL> extractCPFromProperty() {
+    String classpath = System.getProperty("java.class.path");
```

Comment:
In some rare circumstances, this can be `null`

Reply:
Done